### ISSUE TITLE
Async functions does not propogate errors/rejects

### ISSUE THREAD

**Issue by richarddd** [DelSkayn/rquickjs#106]
Given this JS file, i'm expected that errors should log (console object is added to globals).

```javascript
const error = async () => {
    throw "ERROR 1";
};

const error2 = async () => {
    return Promise.reject("ERROR 2");
};

console.log("hello");

error().catch(console.log);
error2().catch(console.log);
```

Expected output:
```shell
node index.js 
hello
ERROR 1
ERROR 2
```

```shell
qjs index.js 
hello
ERROR 1
ERROR 2
```

Current output:
```shell
cargo run -- index.js
hello
```

Reproducable:
https://github.com/richarddd/rquickjs-promise

**Comment by katyo** [comment]
@richarddd Thank you a lot for all your bug reports!
Could you add corresponding tests and open PR? It will be much helpful for us.

**Comment by richarddd** [comment]
> @richarddd Thank you a lot for all your bug reports!
> Could you add corresponding tests and open PR? It will be much helpful for us.

@katyo oh i wish I could do a PR but i basically stated my rust journey two weeks ago and this bug would be a bit to tricky for me. Thank you for a great library and for your awesome work 🙌

**Comment by DelSkayn** [comment]
Hi thanks for the bug report, I have looked into the problem.

Allowing future to yield will result in the expected behavior.
For now adding the line:
```rust
tokio::task::yield_now().await
```
Should fix the problem.

The bug lies within the `Runtime::idle` function.
Due to a bug, the future returned from this function, always, immediately returns, which caused the program to quit before the promises had the opportunity to run.

Should be fixed in #107.

**Comment by DelSkayn** [comment]
Closing this issue as the bug seems to be fixed with the merged PR.
